app/src/main/java/org/mozilla/fenix/tabstray/browser/RemoveTabUseCaseWrapper.kt

```diff
@@ -7,29 +7,17 @@ package org.mozilla.fenix.tabstray.browser
 import mozilla.components.feature.tabs.TabsUseCases
 import org.mozilla.fenix.GleanMetrics.TabsTray
 
-class SelectTabUseCaseWrapper(
```

Comment:
I am a bit curious about our motivation for unwrapping this logic from a UseCase into the Controller. I think it's fine to do this, but I am also a bit vary about what kind of consideration we take for adding a UseCase since @jonalmeida did initially add this as an UseCase. I imagine the intent of that was this UseCase would be more easily reused more broadly, but it didn't pan out that way.

Aside, I also saw RemoveTabUseCaseWrapper wasn't used anywhere except in its own test. https://searchfox.org/mozilla-mobile/search?q=RemoveTabUseCaseWrapper&path=

Reply:
Ah I saw the RemoveTabUseCaseWrapper was already noted.

Reply:
The intent was to do the same things that were needed in multiple places in one "use case":
1. Send telemetry.
2. Call the specific AC use case.
3. Perform a side-effect as part of a callback.
4. Testability.

This was needed in multiple places before so it made sense to have these as reusable parts. Unwrapping this now might make sense; I haven't gone through this patch to comment on that.

I disliked/found it difficult to maintain a high level of regression-free code during that refactor when everything was multiple large classes (hence why multiple interactors were avoided to isolate each use case for each tab list).

Reply:
Yeah, this is more a matter of unwrapping and removing a pass-through class that was used in one location. The AC use case is now called directly in the controller, and we now have test coverage for that controller function (it previously was not tested).